accord-core/src/main/java/accord/coordinate/CheckOn.java

```diff
@@ -94,23 +94,39 @@ protected AbstractRoute route()
     @Override
     protected boolean isSufficient(Id from, CheckStatusOk ok)
     {
-        return ((CheckStatusOkFull)ok).sufficientFor(route()).compareTo(sufficient) >= 0;
+        KeyRanges rangesForNode = topologies().computeRangesForNode(from);
+        PartialRoute scope = this.route.slice(rangesForNode);
+        return isSufficient(scope, ok);
     }
 
     @Override
-    protected void onDone(Done done, Throwable failure)
+    protected boolean isSufficient(CheckStatusOk ok)
     {
+        return isSufficient(route, ok);
```

Comment:
Should this basically be `UnsupportedOperationException` because `isSufficient` is now overridden?

Reply:
No, we separately validate the final `merged` directly with this variant.